frontend/components/TableContainer/DataTable/TruncatedTextCell/TruncatedTextCell.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useRef, useLayoutEffect } from "react";
+import { uniqueId } from "lodash";
+
+import ReactTooltip from "react-tooltip";
+
+interface ITruncatedTextCellProps {
+  value: string | number | boolean;
+  classes?: string;
+}
+
+const TruncatedTextCell = ({
+  value,
+  classes = "w250",
+}: ITruncatedTextCellProps): JSX.Element => {
+  const ref = useRef<HTMLInputElement>(null);
+
+  const [offsetWidth, setOffsetWidth] = useState(0);
+  const [scrollWidth, setScrollWidth] = useState(0);
+
+  useLayoutEffect(() => {
+    if (ref?.current !== null) {
+      setOffsetWidth(ref.current.offsetWidth);
+      setScrollWidth(ref.current.scrollWidth);
+    }
+  }, []);
+
+  const id = uniqueId();
```

Comment:
I believe lodash `uniqueId` is just an incrementing id not a uuid. I don't recall if it would reset to 1 for each instance of the component mounted (or possibly even with each render of the component).

Reply:
Changed to uuidv4 and also changed to uuidv4 for PillCell.tsx to be consistent.